update-common/src/artifacts/update_plan.rs

```diff
@@ -419,11 +420,13 @@ impl<'a> UpdatePlanBuilder<'a> {
         let (artifact_id, bootloader_caboose) =
             read_hubris_caboose_from_archive(artifact_id, data.clone())?;
 
+        let sign = bootloader_caboose.sign.expect("required SIGN in caboose");
```

Comment:
I think we should return an appropriate error here instead of panicking. Development images may be missing their `SIGN` fields, and that shouldn't be a fatal error.